src/plugins/intel_gpu/src/graph/select.cpp

```diff
@@ -31,6 +33,28 @@ layout select_inst::calc_output_layout(select_node const& node, kernel_impl_para
     return layout(in_layout.data_type, in_layout.format, output_size);
 }
 
+template<typename ShapeType>
+std::vector<layout> select_inst::calc_output_layouts(const select_node& /*node*/, const kernel_impl_params& impl_param) {
+    auto input0_layout = impl_param.get_input_layout(0);
+    auto input1_layout = impl_param.get_input_layout(1);
+    auto input2_layout = impl_param.get_input_layout(2);
+
+    auto desc = impl_param.typed_desc<select>();
+    auto dt = desc->output_data_type.value_or(input1_layout.data_type);
+
+    ov::op::v1::Select op;
+    std::vector<ShapeType> output_shapes = { ShapeType{} };
+    std::vector<ShapeType> input_shapes = {
+        input0_layout.get_partial_shape(),
+        input1_layout.get_partial_shape(),
+        input2_layout.get_partial_shape()
+    };
+
+    ov::op::v1::shape_infer(&op, input_shapes, output_shapes);
+
+    return {{output_shapes[0], dt, format::get_default_format(output_shapes[0].size())}};
```

Comment:
Overall looks good, but I think we need to add some handle of dynamic shape case. Trying to access ".size()" for  dynamic shape will cause the error. Please add some dynamic input case for the unittest and handle it here.

Reply:
Why? `output_shapes[0].size()` will return rank of tensor which is supposed to be static (as we don't support dyn rank at the moment).

Reply:
Yes we dont support dyn rank, but shouldnt such a case be detected? (i.e., the assertion in rank().is_static() in size() shouldnt be checked?)